api/include/opentelemetry/logs/logger.h

```diff
@@ -51,45 +124,75 @@ class Logger
    * @param timestamp the timestamp the log record was created.
    * @throws No exceptions under any circumstances.
    */
-
-  /**
-   * The base Log(...) method that all other Log(...) overloaded methods will eventually call,
-   * in order to create a log record.
-   */
   virtual void Log(Severity severity,
                    nostd::string_view body,
                    const common::KeyValueIterable &attributes,
                    trace::TraceId trace_id,
                    trace::SpanId span_id,
                    trace::TraceFlags trace_flags,
-                   common::SystemTimestamp timestamp) noexcept = 0;
+                   common::SystemTimestamp timestamp) noexcept
+  {
+    nostd::unique_ptr<LogRecord> log_record = CreateLogRecord();
```

Comment:
Please confirm if the existing methods wouldn't use LogRecord, but directly use Recordable interface ?

Reply:
`Recordable` derive from `LogRecord`. It's a break change in API here, the existing old `UserLogger::Log` must be replaced by `UserLogger::CreateLogRecord` and `UserLogger::EmitLogRecord`, or it won't compile.(Just like the changes in `etw::Logger`).
And the existing exporters or processes use the old `Recordable` still work after add a new override `Recordable::SetObservedTimestamp` function.

Reply:
Sorry for the delay. Let me go through all the changes in next couple of days, and would complete the review by then.

Reply:
Thanks, please let me know if there is any misunderstanding about specification.

Reply:
No, I think the changes are much aligned to the specs, and look better. I would have some minor comments only.